examples/cli/debug.rs

```diff
@@ -49,6 +65,106 @@ pub async fn debug_group_messages(
     Ok(())
 }
 
+pub async fn debug_welcome_messages(
+    client: &Client<Box<dyn XmtpApi>>,
+    installation_id: Vec<u8>,
+) -> Result<(), String> {
+    let api_client = client.api();
+    let envelopes = api_client
+        .query_welcome_messages(installation_id, None)
+        .await
+        .unwrap();
+    for envelope in envelopes {
+        let msgv1 = match &envelope.version {
+            Some(WelcomeMessageVersion::V1(value)) => value,
+            _ => return Err("Invalid welcome message version".to_string()),
+        };
+        let body = match MlsMessageIn::tls_deserialize_exact(&msgv1.data)
+            .map_err(|e| e.to_string())?
+            .extract()
+        {
+            MlsMessageBodyIn::PrivateMessage(message) => message,
+            _ => return Err("Unsupported message type".to_string()),
+        };
+        let timestamp = format_timestamp(msgv1.created_ns);
+        let sequence_id = msgv1.id;
+        let epoch = body.epoch().as_u64();
+        let content_type = body.content_type();
+        info!("[{timestamp}] [Epoch {epoch}] [Seq {sequence_id}] {content_type:?}");
+    }
+
+    Ok(())
+}
+
+pub async fn debug_key_packages(
+    client: &Client<Box<dyn XmtpApi>>,
+    installation_id: Vec<u8>,
+) -> Result<(), String> {
+    let api_client = client.api();
+
+    let key_package_results = api_client
+        .fetch_key_packages(vec![installation_id])
+        .await
+        .unwrap();
+
+    let mls_provider = client.mls_provider().unwrap();
+
+    let envelopes: Result<Vec<VerifiedKeyPackageV2>, _> = key_package_results
+        .values()
+        .map(|bytes| VerifiedKeyPackageV2::from_bytes(mls_provider.crypto(), bytes.as_slice()))
+        .collect();
+
+    for envelope in envelopes.unwrap() {
+        let inbox_id = envelope.credential.inbox_id;
+        let pkey = hex::encode(envelope.installation_public_key);
+        info!("[InboxId] {inbox_id} [Key Packages] {pkey}");
```

Comment:
Just a nit, but the formatting of these messages looks different than the others, which enclose the key and the value in a `[ ]`

Reply:
fixed